ci/scripts/python_sdist_build.sh

```diff
@@ -23,5 +23,5 @@ source_dir=${1}/python
 
 pushd ${source_dir}
 export SETUPTOOLS_SCM_PRETEND_VERSION=${PYARROW_VERSION:-}
-${PYTHON:-python} setup.py sdist
+${PYTHON:-python} -m build --sdist .
```

Comment:
We're not using Meson here?

Reply:
This is the documented way to create a source distribution via meson-python. See also https://mesonbuild.com/meson-python/how-to-guides/sdist.html

Reply:
Isn't this being run in a checkout anyway? (in ci etc.), I assume you saw failures somewhere?

Reply:
The failure was happening during the release verification script, which was testing sdist creation outside of the git VCS